message-analyzer/Startup.cs

```diff
@@ -0,0 +1,110 @@
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Startup class.
+    /// </summary>
+    public class Startup
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        public const string PubsubTopicName = "receivemediapost";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public const string BindingName = "messagebinding";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        public Startup(IConfiguration configuration)
+        {
+            this.Configuration = configuration;
+        }
+
+        /// <summary>
+        /// Gets the configuration.
+        /// </summary>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// Configures Services.
+        /// </summary>
+        /// <param name="services">Service Collection.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddDaprClient();
+
+            services.AddSingleton(new JsonSerializerOptions()
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                PropertyNameCaseInsensitive = true,
+            });
+        }
+
+        /// <summary>
+        /// Configures Application Builder and WebHost environment.
+        /// </summary>
+        /// <param name="app">Application builder.</param>
+        /// <param name="env">Webhost environment.</param>
+        /// <param name="serializerOptions">Options for JSON serialization.</param>
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, JsonSerializerOptions serializerOptions)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseRouting();
+
+            app.UseCloudEvents();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapSubscribeHandler();
+
+                endpoints.MapPost(PubsubTopicName, ReceiveMediaPost).WithTopic(PubsubTopicName);
+            });
+
+            // Receive a "Post" object from the previous app in the pipeline.
+            async Task ReceiveMediaPost(HttpContext context)
+            {
+                Console.WriteLine("Enter ReceiveMediaPost");
+                var client = context.RequestServices.GetRequiredService<DaprClient>();
+
+                var data = await JsonSerializer.DeserializeAsync<Post>(context.Request.Body, serializerOptions);
+
+                PostWithSentiment postWithSentiment = new PostWithSentiment(data)
+                {
+                    Sentiment = GenerateRandomSentiment()
+                };
+
+                await client.InvokeBindingAsync<PostWithSentiment>(BindingName, postWithSentiment);
+            }
+        }
+
+        internal int GenerateRandomSentiment()
```

Comment:
Can we have a different distribution for each sentiment?

Something like:
Positive: 70%
Neutral: 20%
Negative: 10%

If we have 5 starts, then I would say it would start more like a normal distribution, with peak at 3 or 4. This is just to make the output look legit.

Reply:
Would like to do this later.